modmesh/app/linear_wave.py

```diff
@@ -0,0 +1,103 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+import numpy as np
+
+from matplotlib.backends.qt_compat import QtWidgets
+from matplotlib.backends.backend_qtagg import (
+    FigureCanvas, NavigationToolbar2QT as NavigationToolbar)
+from matplotlib.figure import Figure
+
+from .. import view
+from .. import spacetime as libst
+
+
+def load_app():
+    view.app.pytext.code = """import importlib
+from modmesh import spacetime as libst
+from modmesh.app import linear_wave as app
+
+svr = app.run_linear()
+"""
+
+
+class ApplicationWindow(QtWidgets.QMainWindow):
+    def __init__(self):
+        super().__init__()
+        self._main = QtWidgets.QWidget()
+        self.setCentralWidget(self._main)
+        layout = QtWidgets.QVBoxLayout(self._main)
+
+        static_canvas = FigureCanvas(Figure(figsize=(15, 10)))
+        # Ideally one would use self.addToolBar here, but it is slightly
+        # incompatible between PyQt6 and other bindings, so we just add the
+        # toolbar as a plain widget instead.
+        layout.addWidget(static_canvas)
+        layout.addWidget(NavigationToolbar(static_canvas, self))
+
+        self.ax = static_canvas.figure.subplots()
+
+
+_ui_holder = []
+
+
+def run_linear():
+    grid = libst.Grid(0, 4 * 2 * np.pi, 4 * 64)
+
+    cfl = 1
+    dx = (grid.xmax - grid.xmin) / grid.ncelm
+    dt = dx * cfl
+    svr = libst.LinearScalarSolver(grid=grid, time_increment=dt)
+
+    # Initialize
+    for e in svr.selms(odd_plane=False):
+        if e.xctr < 2 * np.pi or e.xctr > 2 * 2 * np.pi:
+            v = 0
+            dv = 0
+        else:
+            v = np.sin(e.xctr)
+            dv = np.cos(e.xctr)
+        e.set_so0(0, v)
+        e.set_so1(0, dv)
+
+    win = ApplicationWindow()
+    win.show()
+    win.activateWindow()
+    # FIXME: This is a hack for PySide object lifecycle management.
```

Comment:
I have not fully understand how PySide manages the lifecycle of the Qt objects it wraps.  If I do not keep the object, it will go away and crashes.